pkg/constants/constants.go

```diff
@@ -52,7 +52,7 @@ var (
 // Controller Constants
 var (
 	ControllerLabelName        = KFServingName + "-controller-manager"
-	DefaultTimeout       int64 = 300
+	DefaultTimeout       int64 = 30
```

Comment:
Can we set this to 0?

Reply:
this is request timeout, maybe we should expose on our deployment spec ?

Reply:
I see KFServing's role to take as many "ML opinions" as we can. This seems like a perfect case where we can take the ML workload expectation of statelessness. If this assumption is broken at some point, I'd be happy to expose it.

We can also always be override via an Annotation.

Reply:
I feel 30 is safer default here (considering K8S uses same value) since potentially user-container might indeed has some graceful shutdown after receiving SIGTERM, especially for custom model type.

Reply:
300s is the default value(https://github.com/knative/serving/blob/master/config/config-defaults.yaml#L43) from Knative, the fact that Knative supports scaling down to 0 makes possible that some requests may take that long to process, the request is buffered at activator and waits the pod to scale up back from 0.

Reply:
@yuzisun what would be the issue if we set `revision-timeout-seconds` too low?

Reply:
The thing is, knative re-use `revision-timeout-seconds` for pod's `terminationGracePeriod` (https://github.com/knative/serving/blob/fa89c5a2d73e96e09e282ef335d1df1748b53a73/pkg/reconciler/revision/resources/deploy.go#L162)

Reply:
yes, when pods are shut down it needs to wait for all the in-flight requests to get chance to finish processing within the specified request timeout period, so I think it makes sense to re-use revision-timeout-seconds for pod's terminationGracePeriod.